### DISCUSSION TITLE
Should `implicit_authorization_target` be a public method?

### DISCUSSION THREAD

**Discussion by brendon** [palkan/action_policy#183] [Q&A]
I noticed that `implicit_authorization_target` is a public method in `action_policy` on the controller.

Does it need to be public? It feels a bit wrong to have it as a public method on my controllers as that's the convention to use for declaring controller actions (though that's just a hangup from early Rails where public methods were inherently actions on controllers - not the case anymore with restful routes).

What are your thoughts @palkan?

**Comment by palkan** [comment]
> Does it need to be public?

Not necessary; we only use it in the context of `self`, so it could be private as well.

The reason why it's public now is that we add via including a module; and dealing with visibility in that case (i.e., adding `private :method` in `included`) introduces unnecessary complexity to the source code.

**Comment by brendon** [comment]
Thanks @palkan.

I notice you're using `extend ActiveSupport::Concern`. My understanding of that is that any instance methods declared beneath `private` in this module will be private on the including controller: https://stackoverflow.com/a/33768314/129798

So you could just do:

```
private

def implicit_authorization_target
  controller_name.classify.safe_constantize
end
```
